### ISSUE TITLE
HTTP output plugin with JSON format selected does not produce valid JSON

### ISSUE THREAD

**Issue by brendanhill** [fluent/fluent-bit#247]
Hello,

Maybe I am missing something, but I am unable to use the http output plugin to send an apache log that is already json formatted. I am probably doing something improper, and am unsure where to proceed from here.

**fluent-bit config**:
[INPUT]
    Name  tail
    Tag   tag
    Path /var/log/httpd/json_log

[OUTPUT]
    Name  http
    Match *
    Format json
    Host  127.0.0.1
    Port  11235
    URI   /path

[OUTPUT]
    Name stdout
    Match *

**actual apache log entry**:
{"e_id":"dev.111.system.apache.web", "e_time":"2017-04-25T10:12:1493140372Z", "time":"[25/Apr/2017:10:12:52 -0700]", "remoteIP":"172.25.181.130", "host":"10.36.159.204", "request":"/", "query":"", "method":"GET", "status":"403", "userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36", "referer":"-"}

**fluent-bit stdout**:
[0] tag: [1493142935, {"log"=>"{"e_id":"dev.111.system.apache.web", "e_time":"2017-04-25T10:12:1493140372Z", "time":"[25/Apr/2017:10:12:52 -0700]", "remoteIP":"172.25.181.130", "host":"10.36.159.204", "request":"/", "query":"", "method":"GET", "status":"403", "userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36", "referer":"-"}"}]

**ngrep capture** (ngrep -W none -q '.' 'tcp port 11235' -d lo):
[{"date":1493142935, "log":**"**{"e_id":"dev.111.system.apache.web", "e_time":"2017-04-25T10:12:1493140372Z", "time":"[25/Apr/2017:10:12:52 -0700]", "remoteIP":"172.25.181.130", "host":"10.36.159.204", "request":"/", "query":"", "method":"GET", "status":"403", "userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/57.0.2987.133 Safari/537.36", "referer":"-"}**"**}]

**apache log format**:
LogFormat "{\"e_id\":\"dev.111.system.apache.web\", \"e_time\":\"%{%Y-%m-%dT%H:%M:%sZ}t\", \"time\":\"%t\", \"remoteIP\":\"%a\", \"host\":\"%V\", \"request\":\"%U\", \"query\":\"%q\", \"method\":\"%m\", \"status\":\"%>s\", \"userAgent\":\"%{User-agent}i\", \"referer\":\"%{Referer}i\"}" jsonapache

**webservice error**:
ValueError: Expecting , delimiter: line 1 column 30 (char 30)

It seems that the value for the 'log' key is double quoting the JSON object (quotes in bold in the ngrep sample). Maybe I should not be using a json log file, but am unable to find documentation on how to use the apache2 paser either (http://fluentbit.io/documentation/0.11/parser/README.md). 

I apologize in advance if this seems trivial, or I am missing something basic...

**Comment by edsiper** [comment]
By default Fluent Bit will take the original log entry and compose it in a structured format. If no format is given the content will be placed inside a "log" : "your log string". 

If your log is already in JSON, edit your parsers.conf file and append something like this:

```
[PARSER]
    Name        apache_json
    Format      json
```

then in your fluent-bit.conf, specify the new parser in the tail section, e.g:

```
[INPUT]
    Name    tail
    Tag        tag
    Path      /var/log/httpd/json_log
    Parser   apache_json
```

the only restriction in 0.11 series is that the JSON parser cannot extract the time field (this is solved in 0.12-dev)

**Comment by brendanhill** [comment]
Thank you so much for the follow up, that was exactly the information I needed.  I was unaware how the parser came into play.  I will close this as well. Keep up the great work.
